test-network-function/networking/suite.go

```diff
@@ -275,3 +299,101 @@ func testNodePort(env *config.TestEnvironment) {
 		}
 	})
 }
+
+func getDeclaredPortList(command string, container int, podName, podNamespace string) []PortDeclared {
+	var result []PortDeclared
+	protocolName, port, name := "", "", ""
+	ocCommandToExecute := fmt.Sprintf(command, podName, podNamespace, container)
+	res, _ := utils.ExecuteCommand(ocCommandToExecute, ocCommandTimeOut, interactive.GetContext(false))
+	x := strings.Split(res, "\n")
+	for _, i := range x {
+		if strings.Contains(i, "containerPort") {
+			s := strings.Split(i, ":")
+			s = strings.Split(s[1], ",")
+			port = s[0]
+		}
+		if strings.Contains(i, "name") {
+			s := strings.Split(i, ":")
+			s = strings.Split(s[1], ",")
+			name = s[0]
+		}
+		if strings.Contains(i, "protocol") {
+			s := strings.Split(i, ":")
+			protocolName = s[1]
+			continue
+		}
+		if !(port == "" && name == "" && protocolName == "") {
+			continue
+		}
+		p, _ := strconv.Atoi(strings.TrimSpace(port))
+		noderes := PortDeclared{port: p, name: name, protocol: protocolName}
+		result = append(result, noderes)
+	}
+	return result
+}
+
+func getListeningPortList(ocCommand string) []PortListen {
+	res, _ := utils.ExecuteCommand(ocCommand, ocCommandTimeOut, interactive.GetContext(false))
+	lines := strings.Split(res, "\n")
+	var result []PortListen
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if indexProtocolName > len(fields) || indexPort > len(fields) {
+			return result
+		}
+		s := strings.Split(fields[indexPort], ":")
+		p, _ := strconv.Atoi(s[1])
+		nodeListen := PortListen{port: p, protocol: fields[indexProtocolName]}
+		result = append(result, nodeListen)
```

Comment:
```suggestion
		result = append(result, PortListen{port: p, protocol: fields[indexProtocolName]})
```

Reply:
finally I was use in map and not use in struct.